### PR TITLE
Fix two bugs with --logging_first_step

### PR DESCRIPTION
**Pull request by abisee**
# What does this PR do?
This PR fixes two bugs relating to the `--logging_first_step` flag:
1. Though the description for `--logging_first_step` says `"Log and eval the first global_step"`, the flag doesn't actually eval (it only logs). This PR makes sure that eval happens on the first step.
2. When `--logging_first_step` is on, the logged training loss for the first step is miscalculated in `Trainer._maybe_log_save_evaluate`:
```python
logs["loss"] = (tr_loss_scalar - self._logging_loss_scalar) / self.args.logging_steps
```
This divides the loss by `logging_steps` (which is typically large, e.g. 500), when it should be divided by 1. This PR makes sure that the loss is divided by the correct number of steps.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to the it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@sgugger

### CODE REVIEW

**Comment by abisee** [comment]
OK, both the description and the behavior is now logging only (no eval).

**Comment by sgugger** [comment]
Perfect, thanks!

**Review by LysandreJik — APPROVED** [review]
LGTM! Thanks @abisee!
